### PR TITLE
add while, take-while, drop-while

### PR DESCRIPTION
**Pull request by zachcp**
and some simple tests for them

### CODE REVIEW

File: pixie/stdlib.pxi [hunk 1]
```diff
@@ -1455,6 +1455,63 @@ The new value is thus `(apply f current-value-of-atom args)`."
...
+    `(loop []
+      (when ~test
+        ~@body
+        (recur))))
+
+(defn take-while
+  {:doc "Returns a lazy sequence of successive items from coll while
+        (pred item) returns true. pred must be free of side-effects.
+        Returns a transducer when no collection is provided."
+  :added "10.1"}
```
Comment (halgari) [inline pixie/stdlib.pxi]: 0.1 perhaps?

**Comment by gigasquid** [comment]
Nice.  My only comment is that since the code is handling transducers, we should have some tests for them.

**Comment by zachcp** [comment]
@gigasquid sure. just not really sure how to do that.....  call with no arity and see if it returns a functions.

something like this maybe?:
(t/assert=    (type (take-while pos?)) pixie.stdlib.Closure)

**Comment by heyLu** [comment]
You can test them by using `transduce` directly, using something like this: `(transduce (take-while even?) conj [2 4 6 7 8])`.

**Comment by zachcp** [comment]
thanks @heyLu  I added the transduce examples to the tests.

**Comment by zachcp** [comment]
added more tests to check trandsuce when there is an initial value.

**Comment by zachcp** [comment]
updated the drop-while to avoid multiple lookups/atom resets once the drop condition has been reached. idea/code courtesy of  @lgrapenthin

**Pull request merged**